Feature/document upload popup

arpita739 opened the pull request:
PR #154 

Acceptance criteria
Criterion 1: When a document is uploaded the Popup opens automatically - Done ✅
Criterion 2: The Popup features 2 dropdowns - one for Category and one for Building - Done ✅
Criterion 3: Building dropdown display list of all available buildings and a "No Building" (null) option - Done ✅
Criterion 4: Document category dropdown displays all document categories, a "No Category" (null) option and a manual input option (e.g. "Other...:" with an inline input field) - Done  ✅
Criterion 5: If the response data of uploading the document contains a docCategory field, it is preselected in the Category dropdown, else "No Category" is preselected - Done ✅ _Code Implemented_
Criterion 6: If the response data of uploading the document contains a building field, it is preselected in the Building dropdown, else "No Building" is preselected - Done ✅ _Code Implemented_
Criterion 7: On clicking submit the documents building and category is updated in the database (using OpenAPI client) and the popup closes -  #154 ⚠️
Criterion 8: An error / success message is shown - Done ✅
<img width="372" alt="Screenshot 2025-06-16 at 6 17 39 PM" src="https://github.com/user-attachments/assets/f7d01013-b94c-4f9c-becb-225e09611909" />
<img width="340" alt="Screenshot 2025-06-16 at 6 17 49 PM" src="https://github.com/user-attachments/assets/500386a0-2d2e-4fae-9da8-266035b72a8b" />
<img width="512" alt="Screenshot 2025-06-16 at 6 18 07 PM" src="https://github.com/user-attachments/assets/d5adcbd5-5c96-4b6f-ac20-f0a37f8d6071" />
<img width="494" alt="Screenshot 2025-06-16 at 6 18 27 PM" src="https://github.com/user-attachments/assets/4765a2ad-d561-4f81-b07c-f65f9c9c10f9" />

Criterion 7 ⚠️
> In onSave() the component calls updateDocumentMetadata(), which invokes categoryService.assignDocumentCategory(documentId, categoryId, buildingId || 0).
> categoryService is OpenAPI-generated client, so the document’s category and building are persisted to the backend.
> On successful response the component: Emits a saveMetadata event to inform parent components. Shows a success notification. Calls  onClose() (after a 1.5 s delay) to close the popup. Thus, clicking Submit updates the database and then closes the dialog.

kamsur commented:
Criteria 1 successful
Criteria 2 successful
Criteria 3 successful but the document has already been uploaded before user can create new building, come back and reupload document with correct options
Criteria 4 successful but the new category created by inline input shows up twice in dropdown. And again document has already been uploaded before user can choose and save category
Criteria 5&6 are probably failing. No buildingId, categoryId are being set because the document is sent to backend even before the user interacts with popup and saves the options. So no matter what options are chosen they will not be reflected in backend. so criteria 7 is also failing. I have not seen the code for this only the results in UI and backend through swagger.
Criteria 8 i am not sure because i do not see any notification banner and am directly redirected to document view. Please provide any screenshots for reference as to what i should expect

arpita739 commented:
> Criteria 1 successful Criteria 2 successful Criteria 3 successful but the document has already been uploaded before user can create new building, come back and reupload document with correct options Criteria 4 successful but the new category created by inline input shows up twice in dropdown. And again document has already been uploaded before user can choose and save category Criteria 5&6 are probably failing. No buildingId, categoryId are being set because the document is sent to backend even before the user interacts with popup and saves the options. So no matter what options are chosen they will not be reflected in backend. so criteria 7 is also failing. I have not seen the code for this only the results in UI and backend through swagger. Criteria 8 i am not sure because i do not see any notification banner and am directly redirected to document view. Please provide any screenshots for reference as to what i should expect @arpita739

Thanks for the review working on it.

arpita739 commented:
I've made several changes to fix the issues with the document metadata popup functionality:

1. Fixed Document Upload Workflow

The main issue was that the document metadata was being saved before the user had a chance to select categories and buildings. I've modified the upload process to:
User selects a file
Metadata popup appears for category/building selection
After metadata selection, file is uploaded and then metadata is assigned
User receives feedback and is redirected to view the document


8. <img width="1280" alt="Screenshot 2025-06-10 at 12 22 56 AM" src="https://github.com/user-attachments/assets/96ad8f2d-27fb-4ca1-a5c6-7dd1af492a8f" />
